### PR TITLE
[12.0][ADD] product_lot_sequence

### PR DESCRIPTION
**Pull request by AdriaGForgeFlow**
Adds ability to define a lot sequence from the product which will be proposed upon creating new lots.

To use this module:

* Go to Inventory > Products
* Under the inventory tab when tracking set to lots, 4 new fields will be displayed
* If a sequence is not selected, a new one using the prefix, padding and next number fields, will be created
* Go to Inventory > Lot/Serial Numbers
* Create new Lot/Serial number
* Select the product and the next number of the product sequence will be automatically proposed

cc ~ @Eficent

### CODE REVIEW

File: product_lot_sequence/models/product.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
+# Copyright 2019 Eficent Business and IT Consulting Services S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    sequence_id = fields.Many2one(
```
Comment (rousseldenis) [inline product_lot_sequence/models/product.py]: Please change to 'lot_sequence_id'.

Same for all fields under. Your field names are too generic

File: product_lot_sequence/models/product.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    )
+    sequence_number_next = fields.Integer(
+        string='Next Number',
+        help='The next sequence number will be used for the next lot.',
+        compute='_compute_seq_number_next',
+        inverse='_inverse_seq_number_next'
+    )
+
+    @api.model
+    def _create_sequence(self, vals):
```
Comment (rousseldenis) [inline product_lot_sequence/models/product.py]: Same remark for function name

**Review by rousseldenis — CHANGES_REQUESTED** [review]
LGTM. Please adapt all your variables and functions when they are in a non specific model (product.template here)

**Comment by rousseldenis** [comment]
@AdriaGForgeFlow

File: product_lot_sequence/models/product.py (hunk 1 above)
Comment (AdriaGForgeFlow) [reply product_lot_sequence/models/product.py]: Done :+1:

File: product_lot_sequence/models/product.py (hunk 2 above)
Comment (AdriaGForgeFlow) [reply product_lot_sequence/models/product.py]: Done :+1:

**Comment by AdriaGForgeFlow** [comment]
@rousseldenis Comments attended. Could you please update the review? Thanks!

**Review by rousseldenis — APPROVED** [review]

**Review by HviorForgeFlow — APPROVED** [review]

**Review by LoisRForgeFlow — APPROVED** [review]
LGTM

**Comment by LoisRForgeFlow** [comment]
@rousseldenis feel free to merge!

**Pull request merged**